fdbclient/TestKnobCollection.cpp

```diff
@@ -56,23 +56,36 @@ bool TestKnobCollection::trySetKnob(std::string const& knobName, KnobValueRef co
 	return serverKnobCollection.trySetKnob(knobName, knobValue) || knobValue.visitSetKnob(knobName, testKnobs);
 }
 
-#define init(knob, value) initKnob(knob, value, #knob)
+bool TestKnobCollection::isAtomic(std::string const& knobName) const {
+	return serverKnobCollection.isAtomic(knobName) || testKnobs.isAtomic(knobName);
+}
+
+#define init(knob, value, atomic) initKnob(knob, value, #knob, atomic)
```

Comment:
`ServerKnobs.cpp`, `ClientKnobs.cpp`, and `flow/Knobs.cpp` should use this same macro instead of the ones currently defined in this PR.

Reply:
The issue is that the macros in `ServerKnobs.cpp`, `ClientKnobs.cpp`, and `flow/Knobs.cpp` exist to allow the `Atomic` parameter to be optional, so a knob can be specified like
```
init(KNOB_NAME, KNOB_VALUE)
```
or
```
init(KNOB_NAME, KNOB_VALUE, Atomic::NO)
```
Macro overloading requires the "hack" in the PR. There are a few options I can think of here, because I agree that repeating the macro overload definitions in every knob file is a bit ugly.

1. I could move the macro overloads to one header file and include it in all the knob files.
2. I could add an explicit `Atomic` value to each knob line, so `init` always has three parameters. This would definitely be more explicit in terms of knob atomicity, but it is also more redundant and requires a lot more modification to the source.

Reply:
I added a commit that moves most of the helper macros to `flow/Knobs.h`, to clean up some of the repeated code.

Reply:
That makes sense, I think the current implementation is fine.